src/FormContainer.tsx

```diff
@@ -23,13 +25,18 @@ const makeWrapper = <T extends {}>(config: IFormConfig<T>) => (WrappedComponent:
         };
 
         setProperty = (prop: keyof T, value: any) =>
-            this.setState((prevState: any) => ({
-                ...prevState,
-                model: {
-                    ...prevState.model,
-                    [prop]: value
-                }
-            }));
+            this.setState((prevState: any) => {
+                return {
+                    ...prevState,
+                    model: {
+                        ...prevState.model,
+                        [prop]: value
+                    },
+                    ...(prevState.submitErrors[prop] && {
+                        submitErrors: omit(prevState.submitErrors, prop)
```

Comment:
potentially this can be moved to a clearSubmitErrors method

Reply:
Done!